src/transformers/generation_utils.py

```diff
@@ -822,6 +840,7 @@ def generate(
 
         if self.config.is_encoder_decoder:
             # add encoder_outputs to model_kwargs
+            encoder_input_ids = input_ids
```

Comment:
(nit) the comment above this line should go under this line. And maybe we can add a short comment as well. Like "need to keep a pointer to `input_ids` when `encoder_no_repeat_ngram_size` is used.

Reply:
And I think we should do `encoder_input_ids = input_ids if self.config.is_encoder_decoder else None` outside of this if-statement to make the tests pass

Reply:
Done !